docs/_blog/bdk_rn_making_of.md

```diff
@@ -0,0 +1,238 @@
+
+```kotlin
+import com.facebook.react.bridge.*
+```
+
+A new class needs to be defined here which will implement the React Context API
+
+```kotlin
+class BdkRnModule(reactContext: ReactApplicationContext) :
+	ReactContextBaseJavaModule(reactContext) {
+      override fun getName() = "BdkRnModule"
+  }
+```
+
+With the base imports and class defined, we can start writing methods.
+This will demonstrate how bdk native module will be called and how values will be returned to JS over the native bridge
+
+Lets create a method that can be called from JaveScript, to do so we use the `@ReactMethod` directive which is part of the React Context API. This will expose the method so that it can be called from JavaScript.
+
+```kotlin
+@ReactMethod
+fun createWallet(result: Promise) {
+  
+}
+```
+
+We need one more file to complete our basic native framework. A new Kotlin file, `BdkRnPackage.kt` is required to package all our native code into a new android module. This can be done by adding the following code:
+
+```kotlin
+// File: BdkRnPackage.kt
+
+import com.facebook.react.ReactPackage
+import com.facebook.react.bridge.NativeModule
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.uimanager.ViewManager
+
+class BdkRnPackage : ReactPackage {
+
+    override fun createNativeModules(reactContext: ReactApplicationContext):
+            MutableList<NativeModule> {
+        return mutableListOf(BdkRnModule(reactContext))
+    }
+}
+
+```
+
+Now lets add code for creating a wallet in BdkRnModule.kt
+
+The methods used here are for bdk-kotlin and available in the bdk-kotlin documentation.
+
+We first create a key info object
+
+```kotlin
+// File: BdkRnModule.kt
+
+@ReactMethod
+fun createWallet(result: Promise) {
+  // Create key info with a new mnemonic
+  val keys: ExtendedKeyInfo = generateExtendedKey(
+        Network.TESTNET,
+        WordCount.WORDS12,
+        ""
+  )
+  
+  // more code to follow...
+  // create descriptor and change descriptor
+  // create databaseConfig and blockchainconfig
+  // create wallet
+  
+}
+```
+
+Then key info used to create a wallet descriptor and change descriptor:
+
+```kotlin
+val descriptor: String = "wpkh(" + keys.xprv + "/84'/1'/0'/0/*)"
+
+val changeDescriptor: String = descriptor.replace("/84'/1'/0'/0/*","/84'/1'/0'/1/*")
+```
+
+To create a wallet with bdk we need to specify wallet descriptor, network, a database config, blockchain config. We intend to use bitcoin testnet and want to use default memory for data. For bitcoin node we will use a public electrum server. We will need to define these parameters to create a wallet.
+
+```kotlin
+val network = `Network.TESTNET`
+val databaseConfig = DatabaseConfig.Memory
+blockchainConfig =
+      BlockchainConfig.Electrum(
+          ElectrumConfig("ssl://electrum.blockstream.info:60002", null, 5u, null, 10u)
+      )
+```
+
+Once done we can use these parameters to create a bdk wallet using the native android bdk library:
```

Comment:
You might want to change bdk and android here to `BDK` and `Android`.

Reply:
Done ✅